.github/workflows/build-deploy-pudl.yml

```diff
@@ -19,40 +19,41 @@ jobs:
     name: Build Docker image, push to Docker Hub and deploy to a GCE VM
     runs-on: ubuntu-latest
     permissions:
-      contents: read
+      contents: write
       id-token: write
     steps:
       - name: Use pudl-deployment-dev vm and dev branch if running on a schedule
         if: ${{ (github.event_name == 'schedule') }}
         run: |
           echo "This action was triggered by a schedule."
           echo "GCE_INSTANCE=pudl-deployment-dev" >> $GITHUB_ENV
-          echo "GCE_INSTANCE: $GCE_INSTANCE"
-          echo "GITHUB_REF=dev" >> $GITHUB_ENV
-          echo "GITHUB_REF: $GITHUB_REF"
+          echo "BUILD_REF=dev" >> $GITHUB_ENV
 
       - name: Checkout Repository
         uses: actions/checkout@v4
         with:
-          ref: ${{ env.GITHUB_REF }}
+          ref: ${{ env.BUILD_REF }}
+          fetch-depth: 0
```

Comment:
Setting the `fetch-depth` to 0 tells the action to check out the whole repo, rather than a shallow copy of just the `BUILD_REF`.  This is necessary in order to have access to the `nightly` branch for merging inside the running container.  I didn't see a way to only check out the two refs we need, and this works fine and doesn't add much time.

Reply:
This seems fine, though also the suggestion I added below works with a shallow clone on my machine:

```
> git clone https://github.com/catalyst-cooperative/pudl.git --depth 1 
> cd pudl
> git fetch origin nightly:nightly
> git checkout nightly
```

Reply:
Ah okay great. I'll take the suggestion and switch this back to doing a shallow clone.